runtime/syntax/2html.vim

```diff
@@ -1,2068 +1,5 @@
-
-" Add hyperlinks
-if !s:settings.no_links
-  %s+\(https\=://\S\{-}\)\(\([.,;:}]\=\(\s\|$\)\)\|[\\"'<>]\|&gt;\|&lt;\|&quot;\)+<a href="\1">\1</a>\2+ge
-endif
-
-" The DTD
-if !s:settings.no_doc
-  if s:settings.use_xhtml
-    exe "normal! gg$a\n<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Transitional//EN\" \"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd\">"
-  elseif s:html5
-    exe "normal! gg0i<!DOCTYPE html>\n"
-  else
-    exe "normal! gg0i<!DOCTYPE HTML PUBLIC \"-//W3C//DTD HTML 4.01 Transitional//EN\" \"http://www.w3.org/TR/html4/loose.dtd\">\n"
-  endif
-endif
-
-if s:settings.use_xhtml && !s:settings.no_doc
-  exe "normal! gg/<html/e\na xmlns=\"http://www.w3.org/1999/xhtml\"\e"
-endif
-
-" Cleanup
-%s:\s\+$::e
-
-" Restore old settings (new window first)
-"
-" Don't bother restoring foldmethod in case it was syntax because the markup is
-" so weirdly formatted it can take a LONG time.
-let &l:foldenable = s:old_fen
-let &report = s:old_report
-let &title = s:old_title
-let &icon = s:old_icon
-let &paste = s:old_paste
-let &magic = s:old_magic
-let @/ = s:old_search
-let &more = s:old_more
-
-" switch to original window to restore those settings
-exe s:orgwin .. "wincmd w"
-
-if !s:settings.expand_tabs
-  let &l:isprint = s:old_isprint
-endif
-let &l:stl = s:origwin_stl
-let &l:et = s:old_et
-let &l:scrollbind = s:old_bind
-
-" and back to the new window again to end there
-exe s:newwin .. "wincmd w"
-
-let &l:stl = s:newwin_stl
-exec 'resize' s:old_winheight
-let &l:winfixheight = s:old_winfixheight
-
-let &ls=s:ls
-let &eventignore=s:ei_sav
-
-" Save a little bit of memory (worth doing?)
-unlet s:htmlfont s:whitespace
-unlet s:old_et s:old_paste s:old_icon s:old_report s:old_title s:old_search
-unlet s:old_magic s:old_more s:old_fen s:old_winheight
-unlet! s:old_isprint
-unlet s:whatterm s:stylelist s:diffstylelist s:lnum s:end s:margin s:fgc s:bgc s:old_winfixheight
-unlet! s:col s:id s:attr s:len s:line s:new s:expandedtab s:concealinfo s:diff_mode
-unlet! s:orgwin s:newwin s:orgbufnr s:idx s:i s:offset s:ls s:ei_sav s:origwin_stl
-unlet! s:newwin_stl s:current_syntax
-if !v:profiling
-  delfunc s:HtmlColor
-  delfunc s:HtmlFormat
-  delfunc s:CSS1
-  delfunc s:BuildStyleWrapper
-  if !s:settings.use_css
-    delfunc s:HtmlOpening
-    delfunc s:HtmlClosing
-  endif
-  if s:settings.dynamic_folds
-    delfunc s:FoldCompare
-  endif
-
-  if !s:settings.no_progress
-    delfunc s:ProgressBar
-    delfunc s:progressbar.paint
-    delfunc s:progressbar.incr
-    unlet s:pgb s:progressbar
-  endif
-
-  delfunc s:Add_diff_fill
-endif
-
-unlet! s:new_lnum s:diffattr s:difffillchar s:foldfillchar s:HtmlSpace s:diffstyle
-unlet! s:LeadingSpace s:HtmlEndline s:firstfold s:numcol s:foldcolumn
-unlet! s:wrapperfunc_lines s:build_fun_lines
-unlet s:foldstack s:allfolds s:foldId s:settings
-
-let &cpo = s:cpo_sav
-unlet! s:cpo_sav
-
-" Make sure any patches will probably use consistent indent
-"   vim: ts=8 sw=2 sts=2 noet
+lua << EOF
```

Comment:
?
Didn't you say (in [a previous comment](https://github.com/neovim/neovim/pull/27097#issuecomment-1902061805)):
> I think it's OK to keep the syntax file (and even preferable; makes syncing with Vim runtime slightly easier).
The reason why it's `.vim` is to make `:runtime! syntax/2html.vim` work (which is what the docs say you should use).

Reply:
It's OK to keep it _unchanged_ (because we don't maintain it). But my comment forgot that this was supposed to be the entrance point, which makes zero sense (at least in Neovim). In this case, the file should be removed and the docs should be changed.

Again, we want the functionality, not the interface (which is horrible even by legacy standards).

Reply:
(I'm sorry I didn't check this before.)

Reply:
Then it's easier to remove it.

Reply:
Yes, since it's the main script, it _should_ be removed. My earlier comment was wrong, I apologize.